File: spring-sync-core/src/main/java/org/springframework/sync/diffsync/IPersistenceCallbackRegistry.java
package org.springframework.sync.diffsync;

import java.io.Serializable;

public interface IPersistenceCallbackRegistry {
  void addPersistenceCallback(PersistenceCallback<? extends Serializable> persistenceCallback);

  PersistenceCallback<? extends Serializable> findPersistenceCallback(String key);
}

File: sync-core-kotlin/src/commonMain/kotlin/de/sync/core/persistence/IPersistenceCallbackRegistry.kt
package de.sync.core.persistence

import kotlinx.serialization.Serializable

interface IPersistenceCallbackRegistry {
    fun addPersistenceCallback(persistenceCallback: PersistenceCallback<Serializable>)
    fun findPersistenceCallback(key: String): PersistenceCallback<Serializable>
}